src/frequenz/sdk/timeseries/_serialized_ringbuffer.py

```diff
@@ -0,0 +1,77 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Ringbuffer implementation with serialization support."""
+
+import pickle
+from datetime import datetime, timedelta
+from os.path import exists
+from typing import TypeVar
+
+import numpy as np
+
+from frequenz.sdk.timeseries._ringbuffer import OrderedRingBuffer
+
+T = TypeVar("T")
+Container = TypeVar("Container", list, np.ndarray)
+
+ATTRIBUTES_TO_PICKLE = [
+    "_datetime_oldest",
+    "_datetime_newest",
+    "_buffer",
+    "_gaps",
+    "_sampling_period",
+    "_time_range",
+    "_time_index_alignment",
+]
+
+
+class SerializedRingbuffer(OrderedRingBuffer[T, Container]):
+    """Sorted ringbuffer with serialization support."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        buffer: Container,
+        sampling_period: timedelta,
+        path: str,
+        time_index_alignment: datetime = datetime(1, 1, 1),
+        load_from_disk: bool = True,
+    ) -> None:
```

Comment:
1) Maybe instead of load_from_disk: bool = True - we should check if the data in the path are in self._time_range from OrderedRingBuffer?

2) Would we create gap for time between  `_datetime_oldest` and `datetime.now()` ?

Reply:
no gap would be created, only when you push a new value with a timestamp in the future (future from the ringbuffers perspective, that is)